packages/api/src/modules/workspace/controller.ts

```diff
@@ -82,71 +82,61 @@ export class WorkspaceController {
     }
   }
 
-  // todo: implement this by other coins, and use utils of bsafe-sdk
-  async getBalance(req: IGetBalanceRequest) {
+  async fetchPredicateData(req: IGetBalanceRequest) {
+    let reservedCoins: CoinQuantity[] = [];
     try {
       const { workspace } = req;
       const predicateService = new PredicateService();
       const transactionService = new TransactionService();
-      const predicatesBalance = [];
-      let reservedCoins: CoinQuantity[] = [];
 
-      await predicateService
+      const predicates = (await predicateService
+        .filter({ workspace: [workspace.id] })
+        .list()) as Predicate[];
+
+      const balancePromises = predicates.map(async predicate => {
+        const vault = await predicateService.instancePredicate(predicate.id);
```

Comment:
can change this... the instance predicate run an sql by predicate.
- list instanced predicates or send to .instancePredicate the parameter to prevent a new sql query